src/compiler/scala/tools/nsc/Global.scala

```diff
@@ -1129,6 +1129,12 @@ class Global(var currentSettings: Settings, reporter0: LegacyReporter)
     val profiler: Profiler = Profiler(settings)
     keepPhaseStack = settings.log.isSetByUser
 
+    // We hit these checks regularly. They shouldn't change inside the same run, so cache the comparisons here.
+    val isScala212: Boolean = settings.isScala212
```

Comment:
Side issue: Can we reasonably claim to support this setting anymore after the compiler changes related to vararags?

Reply:
i guess it's always best effort